### ISSUE TITLE
Docker should be considered also as a target for Nautobot production environents and the documentation should fully acknowledge and include this

### ISSUE THREAD

**Issue by itheodoridis** [nautobot/nautobot#2369]
### As ...

Isabella - Infrastructure Owner

### I want ...

to be able to define my docker based environment as Production for Nautobot and this should be acknowledged and included in the installation instructions in the documentation.

### So that ...

The only distinction should be the intent not the platform (as opposed to consider docker based as testing/dev and physical/VM to be production environment.
Please also see https://github.com/nautobot/nautobot-docker-compose/issues/49

### I know this is done when...

I can fully adapt my installation according to the documentation to a standardized docker based  environment with the option to use docker-compose for it.
Please see also https://github.com/nautobot/nautobot-docker-compose/issues/49

### Optional - Feature groups this request pertains to.

- [X] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [X] Documentation
- [X] Other (not directly a platform feature)

**Comment by itdependsnetworks** [comment]
We publish helm-charts and use Docker extensively for development environment. Outside of that, maintaining 2 linux flavors, multiple DBs, etc.. this is outside the scope of the project at this point.

**Comment by itheodoridis** [comment]
Hello Ken. I understand your viewpoint (even after 3 years from this) but besides the fact that we have had already some cooperation on this with people from NTC and that they have already produced an outcome on this (based on Poetry with some controversial points but still quite good), there is also the fact that there are quite a few instances already in Production of Nautobot on docker in operational networks, the bigger being the Network for the Greek State, which has received some attention and reposted extensively from NTC on social media. So I still disagree on this and stand by my original point.
I don't see where maintaining versions for linux or the DBs comes into this discussion. You can include or exclude areas into the project as you see fit. But from what I understand NTC is about to turn into a product company (I read about that in a packet pushers post). So maybe turning away from a portion of the market and pushing people towards your competitors is not the best idea.
Up to you of course. However if NTC starts to fall behind on supporting this platform, I will probably have to turn my efforts into migrating to another product. So far I have had no complaints. We 'll see.

**Comment by itdependsnetworks** [comment]
Just trying to be more clear and have better GitHub hygiene versus letting issues sit for years without any response. More than happy to have a call to understand the use case, how this would benefit you etc. Feel free to send me an email or hit me up on Slack and I can bring our head of product to understand.

**Comment by itheodoridis** [comment]
We have had an official call already in the past (in 2024 I believe), I don't think we can make progress on the use case for my own organization. Besides it's not about that. It's about recognizing the space where there is a different scale for automation projects in the European market, it seldom makes sense for European companies to have a network team maintain Linux hosts with full package management (resolving conflicts etc) or Kubernetes distributions. Docker is more of a natural environment for such teams in this area and this is what American Companies (including the big hardware vendors) often get wrong about this market because they are used to a different scale.
It's not about me or my org. It's about the space.
What you wrote can be interpreted in a number of ways, the two most likely being:
- You are just closing the still open github issue (no problem, if it turned into a separate development with another repo, than the issue is as good as resolved in its original state).
- You are closing the issue AND announcing NTC will no longer be supporting (or rather developing, aiding, whatever other verb seems fitting) Nautobot on docker with docker compose/Poetry (this repo: https://github.com/nautobot/nautobot-docker-compose).
Your viewpoint? I respect 100%.
However if you meant the second thing, at some point I will be forced to switch to a different platform and I believe you will see more teams do the same. Whether I learn/get used to use Kubernetes for Nautobot is not the issue. It's that I can't force this down the throat of my colleagues/team mates with a clear conscience.
Yes happy to take this over to slack or DMs even if you prefer, I will not develop this further here.

**Comment by itdependsnetworks** [comment]
Just for clarification to any future readers, we are not changing anything with: https://github.com/nautobot/nautobot-docker-compose. In general, as a practice we have worked with customers that use Docker or Docker Compose as a production system. As a global recommendation though (e.g. without the nuance conversation mentioned above), it is generally not considered enterprise grade.

As for Docker environments in a developer process, we continue to use Docker in over 20 open source projects, internally for dozens if not more customer repos, as well as the aforementioned project.

Going to re-open the ticket, collect some more feedback from PS, and internally ask around how it's being used.

The grounds for the update was solely GitHub hygiene (expect more in the coming weeks) and better transparency, and not a change in direction in how we support Docker and / or the community, which is not different before or after this comment (e.g. this issue was open as we did not support Docker to begin with and still don't.)

Will ping you on Slack on Wednesday, as my Tuesday am is a bit jam packed.

**Comment by erjac77** [comment]
Hello. I just wanted to share my experience. We've been using Nautobot in production on a Podman infrastructure for almost three years. Since the installation guide mentions it, we assumed the [Docker Compose deployment option](https://docs.nautobot.com/projects/core/en/stable/user-guide/administration/installation/#__tabbed_1_2) was supported. You've sown some doubt…

**Comment by itdependsnetworks** [comment]
@erjac77, yes we support the publication of docker containers and you are following our expected pattern, to use those containers in a tool (podman) that we do not document how to configure. 

This github issue was about documenting and thus ensuring we are testing and maintaining instructions. 

Basically, you are good to go.

**Comment by itheodoridis** [comment]
@itdependsnetworks sure no problem, hit me up on Wednesday. Also feel free to close this whenever you think is time.
@esljaz Ken is right on this being open for way too long and I am guessing it's also kind of a metric of how much they are engaging issues and whether they are making progress. This "issue" was never going to be about developing a feature or fixing a bug. It was to engage people in a discussion (not in this thread) so a additional direction of movement could arise and some progress be made on this while acknowledging this space. I think that was reached with the repo I mentioned (although I still have my objections in keeping some ties to the underlying os space through the way poetry is used to maintain the environment but I can't say I have a better suggestion and I do understand the concept, we have discussed enough with Justin Drew). I also think it's not going away soon as there are still a lot of disadvantages in the other two options (os based, kubernetes). I suspect that Kubernetes may become the default if some platforms issues are solved (persistent common storage) and more do not arise (nginx ingress being retired). Until that becomes a reality, I think you can count on docker being used and supported.
But the message needs to be clear and I think we all accomplished that here.
I expect to discuss it a bit more with Ken in private but also remember that we support each other as well in slack and in here.
